bolt-modules/out/spec/functions/out/verbose_spec.rb

```diff
@@ -0,0 +1,207 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt/executor'
+require 'bolt/inventory'
+
+describe 'out::verbose' do
+  let(:executor)  { Bolt::Executor.new }
+  let(:outputter) { stub('outputter', handle_event: nil) }
+
+  let(:inventory) { Bolt::Inventory.empty }
+  let(:target)    { inventory.get_target('target1') }
+  let(:target2)   { inventory.get_target('target2') }
+
+  let(:result)       { Bolt::Result.new(target, message: "ok", action: 'action') }
+  let(:err_result)   { Bolt::Result.new(target2, error: { 'msg' => 'oops' }, action: 'action') }
+  let(:result_set)   { Bolt::ResultSet.new([result, err_result]) }
+  let(:apply_result) { Bolt::ApplyResult.new(target, report: { 'status' => 'changed' }) }
+
+  let(:error)        { Bolt::Error.new("Task 'watermelon' could not be found", 'bolt/apply-prep') }
+  let(:puppet_error) { Puppet::DataTypes::Error.new('Something went terribly, terribly wrong!') }
+  let(:tasks_enabled) {true}
+  let(:resource) { Bolt::ResourceInstance.new(resource_data) }
+  let(:resource_data) do
+    {
+      'target'        => target,
+      'type'          => 'File',
+      'title'         => '/etc/puppetlabs/',
+      'state'         => { 'ensure' => 'present' },
+      'desired_state' => { 'ensure' => 'absent' },
+      'events'        => [{ 'audited' => false }]
+    }
+  end
+
+  around(:each) do |example|
+    executor.subscribe(outputter)
+
+    Puppet[:tasks] = true
```

Comment:
This will ensure tests are always running in 'task mode', i.e. outside apply blocks. The value here should be a `let` that you can toggle to `false` as needed.

Reply:
What would the syntax be for that? I've tried changing `Puppet[:tasks] = true` to `let(:tasks) {true}` on its current line, and moving it right below line 23 and both raise errors. I also tried doing it in a `before(:each)` with no luck

Reply:
Here's an example in the `run_command` spec tests:
https://github.com/puppetlabs/bolt/blob/main/bolt-modules/boltlib/spec/functions/run_command_spec.rb#L15
https://github.com/puppetlabs/bolt/blob/main/bolt-modules/boltlib/spec/functions/run_command_spec.rb#L12
https://github.com/puppetlabs/bolt/blob/main/bolt-modules/boltlib/spec/functions/run_command_spec.rb#L212-L219